### PR TITLE
Fix regression in iloc with boolean list

### LINKED ISSUE #36741: BUG: Setting values with ILOC and list like indexers raises
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import pandas as pd


df = pd.DataFrame({"flag": ["x", "y"], "value": [1, 2]})
df.iloc[[True, False], 1] = df.iloc[[True, False], 1] * 2
print(df)

```

#### Problem description

This worked on 1.0.5 and returned 
```
  flag  value
0    x      2
1    y      2
```

On master this raises:

```
Traceback (most recent call last):
  File "/home/developer/.config/JetBrains/PyCharm2020.2/scratches/scratch_5.py", line 221, in <module>
    df.iloc[[True, False], 1] = x
  File "/home/developer/PycharmProjects/pandas/pandas/core/indexing.py", line 681, in __setitem__
    iloc._setitem_with_indexer(indexer, value)
  File "/home/developer/PycharmProjects/pandas/pandas/core/indexing.py", line 1756, in _setitem_with_indexer
    self._setitem_single_column(ilocs[0], value, pi)
  File "/home/developer/PycharmProjects/pandas/pandas/core/indexing.py", line 1800, in _setitem_single_column
    ser._mgr = ser._mgr.setitem(indexer=pi, value=value)
  File "/home/developer/PycharmProjects/pandas/pandas/core/internals/managers.py", line 532, in setitem
    return self.apply("setitem", indexer=indexer, value=value)
  File "/home/developer/PycharmProjects/pandas/pandas/core/internals/managers.py", line 397, in apply
    applied = getattr(b, f)(**kwargs)
  File "/home/developer/PycharmProjects/pandas/pandas/core/internals/blocks.py", line 923, in setitem
    check_setitem_lengths(indexer, value, values)
  File "/home/developer/PycharmProjects/pandas/pandas/core/indexers.py", line 158, in check_setitem_lengths
    raise ValueError(
ValueError: cannot set using a list-like indexer with a different length than the value

Process finished with exit code 1

```

Was this change of behavior intended?

#### Expected Output

I would expect, that this would work and returns the results from 1.0.5

#### Output of ``pd.show_versions()``

<details>

master

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #36741
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Run tests in ``indexes`` and ``indexing`` locally without failure.

### CODE REVIEW

**Comment by phofl** [comment]
cc @simonjayhawkins @jreback 

Is this early enough to get into 1.1.4?

File: pandas/tests/indexing/test_iloc.py [hunk 1]
```diff
@@ -767,3 +767,11 @@ def test_iloc_setitem_series_duplicate_columns(self):
         )
         df.iloc[:, 0] = df.iloc[:, 0].astype(np.float64)
         assert df.dtypes.iloc[2] == np.int64
+
+
+def test_iloc_setitem_bool_array():
```
Comment (simonjayhawkins) [inline pandas/tests/indexing/test_iloc.py:772]: could also be put somewhere in pandas\tests\frame\indexing\

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @phofl for the PR. I won't comment on the fix and leave that to @jreback and @jbrockmendel

File: pandas/tests/indexing/test_iloc.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/indexing/test_iloc.py]: Yeah, that's the file I looked for there :) Thanks

**Comment by simonjayhawkins** [comment]
@phofl it appears the test got lost in the merge

**Comment by phofl** [comment]
Thanks very much. That was not that clever :)

**Comment by jreback** [comment]
code fix is ok, but this is going to be tricky to backport as this code as changed a bit.

cc @jbrockmendel merge if looks good to you.

File: pandas/tests/frame/indexing/test_setitem.py [hunk 2]
```diff
@@ -229,3 +229,10 @@ def test_setitem_periodindex(self):
         rs = df.reset_index().set_index("index")
         assert isinstance(rs.index, PeriodIndex)
         tm.assert_index_equal(rs.index, rng)
+
+    def test_iloc_setitem_bool_array(self):
```
Comment (jreback) [inline pandas/tests/frame/indexing/test_setitem.py:233]: this is actually a list can you parameterize on 
```[list, np.array]```  e.g.
```
indexer = klass([True, False, False])
df.iloc[indexer, 1] = .....
```

Comment (phofl) [reply pandas/tests/frame/indexing/test_setitem.py]: Good point. Done. Renamed test too.

**Comment by jreback** [comment]
this is fine, thanks @phofl 

not sure how hard the actual backport to 1.1.4 is

**Comment by jreback** [comment]
@meeseeksdev backport 1.1.x

**Comment by jreback** [comment]
yep this needs manual. @phofl @simonjayhawkins

**Pull request merged**